tests/test_retrieve.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 Northwestern University.
+#
+# invenio-analytics-importer is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+import dataclasses
+import json
+
+import pytest
+
+from invenio_analytics_importer.retrieve import (
+    MatomoAnalytics,
+    get_downloads_per_day,
+)
+
+
+@dataclasses.dataclass
+class FakeResponse:
+    """Fake response."""
+
+    json_body: str
+
+    def raise_for_status(self):
+        """Faked raise_for_status."""
+        return False
+
+    @property
+    def text(self):
+        """Faked text."""
+        return self.json_body
+
+    def json(self):
+        """Faked json."""
+        return json.loads(self.json_body)
+
+
+class FakeClient:
+    """Fake httpx client."""
+
+    def __init__(self):
+        """Constructor."""
+        self._response_data = {}
+
+    def set_response(self, criteria, json_body=""):
+        """Register a response."""
+        self._response_data[criteria] = {
+            "json_body": json_body,
+        }
+
+    def get_response(self, criteria):
+        """Get response."""
+        data = self._response_data[criteria]
+        return FakeResponse(**data)
+
+    async def post(self, url, params=None, data=None):
+        """Post."""
+        return self.get_response((url, params["date"]))
+
+
+@pytest.mark.asyncio
+async def test_get_downloads_per_day(running_app, record_factory, tmp_path):
+    days = ["2024-08-01", "2024-08-02"]
```

Comment:
Are the days set here meant to be a from-to range? Or is it each individual day between a range of dates set somewhere else?

Reply:
The idea is to pass a list of days in the YYYY-MM-DD format. Those will be generated from a range but the function consuming those days doesn't care about that.